lib/cn/node_split.c

```diff
@@ -518,33 +528,54 @@ cn_split(struct cn_compaction_work *w)
 
     assert(!list_empty(&w->cw_node->tn_kvset_list));
 
+    wargs = calloc(w->cw_kvset_cnt, sizeof(*wargs));
+    if (!wargs)
+        return merr(ENOMEM);
+
+    wq = cn_get_io_wq(w->cw_tree->cn);
+
     for (i = 0, le = list_first_entry(&w->cw_node->tn_kvset_list, typeof(*le), le_link);
          i < w->cw_kvset_cnt;
          i++, le = list_next_entry(le, le_link)) {
 
-        struct kvset *ks = le->le_kvset;
-        struct kvset_split_res result = { 0 };
+        INIT_WORK(&wargs[i].work, kvset_split_worker);
+        wargs[i].ks = le->le_kvset;
+        wargs[i].split_kobj = &split_kobj;
+        wargs[i].pc = w->cw_pc;
+        kvset_split_res_init(w, &wargs[i].result, i);
+
+        if (!queue_work(wq, &wargs[i].work)) {
+            err = merr(EBUG);
+            break;
+        }
+    }
 
-        kvset_split_res_init(w, &result, i);
+    /* Wait for all the queued kvset split work to finish */
+    flush_workqueue(wq);
```

Comment:
Not now, but we may want to change this as flush-workqueue prevents new work from starting until all the existing work has completed.  Typically I use a condvar for stuff like this, but we can look into that later...

Reply:
Agreed.